balancer-labs/frontend-v2
# Fetch Pool APR's using SDK

# Description

Switch APR fetching from local calculation to using the SDK instead. Update types to use SDK Pool, PoolType and APRBreakdown types. Switch all locations that APR is displayed to work with the new APRBreakdown type.

This requires adding two new PoolType's to the SDK: https://github.com/balancer-labs/balancer-sdk/pull/208, there are some hacks around this which can be removed once that is merged and SDK is updated. Using SDK PoolType was required now that Pool is extended from the SDK Pool type. 

## Type of change


## How should this be tested?

- Compare this branch to production, ensure all APR's are roughly equivalent (there are some differences due to updates in the way the SDK calculates APR's, these are listed here: https://github.com/balancer-labs/balancer-sdk/pull/98
- Ensure that all APR tooltips show accurate info and the values add up to the main value displayed
- Ensure APR's show correctly on individual pool pages and they match the APR on the main pools list. 

## Checklist:
